README.md

```diff
@@ -1,707 +1,1799 @@
+
+**`Since`** 4.1.0, `defaultMemoize` also accepts an options object as its first argument instead of an `equalityCheck` function. The `options` object may contain:
+
+```ts
+type EqualityFn = (a: any, b: any) => boolean
+
+interface DefaultMemoizeOptions {
+  equalityCheck?: EqualityFn
+  resultEqualityCheck?: EqualityFn
+  maxSize?: number
+}
+```
+
+| Name                  | Description                                                                                                                                                                                                                                                                                                                                                                         |
+| :-------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `equalityCheck`       | Used to compare the individual arguments of the provided calculation function. <br /> **`Default`** = `defaultEqualityCheck`                                                                                                                                                                                                                                                        |
+| `resultEqualityCheck` | If provided, used to compare a newly generated output value against previous values in the cache. If a match is found, the old value is returned. This addresses the common <code>todos.map(todo => todo.id)</code> use case, where an update to another field in the original data causes a recalculation due to changed references, but the output is still effectively the same. |
+| `maxSize`             | The cache size for the selector. If greater than 1, the selector will use an LRU cache internally. <br /> **`Default`** = 1                                                                                                                                                                                                                                                         |
+
+> [!WARNING]
+> If `resultEqualityCheck` is used inside `argsMemoizeOptions` it has no effect.
+
+</details>
+
+<details><summary><b>Returns</b></summary>
+
+A memoized function with a `.clearCache()` method attached.
+
+</details>
+
+<details><summary><b>Type parameters</b></summary>
+
+| Name   | Description                                |
+| :----- | :----------------------------------------- |
+| `Func` | The type of the function that is memoized. |
+
+</details>
+
+<details><summary><b><code>Examples</code></b></summary>
+
+###### Using [`createSelector`]
```

Comment:
:hand: Can we tweak these two titles?  Maybe emphasize "Using `defaultMemoize` with `createSelector`", etc.

Reply:
Gladddddllllyyyy :)